### PR TITLE
Added missing docs for Stack

### PR DESCRIPTION
**Pull request by dominikmaeckel**

### CODE REVIEW

File: docs/util/stack/stack.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+1. TOC
+{:toc}
+</details>
+
+
+## Usage
+The [Stack][StackKDoc] class is a kotlin implementation of the common stack collection.
+
+## Creation
+The stack can be created with initial elements as a constructor parameter, either as collection or varargs, or as an empty stack with no constructor parameters.
```
Comment (fabiankluempers) [inline docs/util/stack/stack.md:46]: Maybe specify in which order the initial elements get added.

File: docs/util/stack/stack.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    Table of contents
+  </summary>
+  {: .text-delta }
+1. TOC
+{:toc}
+</details>
+
+
+## Usage
+The [Stack][StackKDoc] class is a kotlin implementation of the common stack collection.
```
Comment (fabiankluempers) [inline docs/util/stack/stack.md:43]: Could specify that this is a utility class, since stacks are a common data structure in board games.

**Review by fabiankluempers — CHANGES_REQUESTED** [review]

File: docs/util/stack/stack.md (hunk 1 above)
Comment (dominikmaeckel) [reply docs/util/stack/stack.md:46]: Thx, forgot about this while writing just that thing in #pushAll().

**Comment by dominikmaeckel** [comment]
In the constructor, the elements get added in reversed order while pushAll pushes them one by one.
Is this intended or should this be changed?

**Comment by fabiankluempers** [comment]
> In the constructor, the elements get added in reversed order while pushAll pushes them one by one.
> Is this intended or should this be changed?

My thought process was, that one defines the stack in the constructor, while ``pushAll`` is just repeated pushing. We could change either if deemed necessary, but that would break the API again between 0.3 and the next release.

**Review by fabiankluempers — APPROVED** [review]

**Pull request merged**